server/ntb/publish/ntb_nitf_multiservice.py

```diff
@@ -24,6 +27,48 @@ def _format_service(self, article):
         except (KeyError):
             pass
 
+    def _format_docdata_with_imatrics(self, article, docdata):
+        super()._format_docdata(article, docdata)
+        imatrics_topics = [
+            s
+            for s in article.get("subject", [])
+            if s.get("scheme") == "imatrics_topic" and s.get("source") == "imatrics"
+        ]
+        for imatrics_topic in imatrics_topics:
+            key_list = etree.SubElement(docdata, "key-list")
```

Comment:
seems like you're making a `key-list` elem for each topic, that should be done probably only once before the cycle

Reply:
Good catch.